ticketSelling.c

```diff
@@ -0,0 +1,171 @@
+#include <stdio.h>
+#include <pthread.h>
+#include <malloc.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <time.h>
+
+#define NUM_QUEUE 10
+
+typedef struct customer {
+	int arrival_time;
+	int custId;
+} customer;
+
+
+typedef struct customerQ {
+	customer *cust;
+	int front, rear;
+} customerQ;
+
+
+typedef struct ticketSell {
+	int next_avail;
+} ticketSell;
+
+customerQ *custQ;
+ticketSell tickets[10];
+
+float H_serve_time[] = {0.1, .2};
+float M_serve_time[] = {.2, .3, .4};
+float L_serve_time[] = {.4, .5, .6, .7};
+
+pthread_cond_t cond = PTHREAD_COND_INITIALIZER;
+pthread_mutex_t mutex = PTHREAD_MUTEX_INITIALIZER;
+
+static int serve_t = 0;
+// seller thread to serve one time slice (1 minute)
+void * sell(char *seller_type)
+{
+
+	while (serve_t < 60)
+	{
+		pthread_mutex_lock(&mutex);
+		//pthread_cond_wait(&cond, &mutex);
+		
+		if(strcmp(seller_type, "H") == 0){
+			float serve_time  = H_serve_time[rand() % 2];
+			printf("Sleeping thread %s for time %f\n", seller_type, serve_time);
+			sleep(serve_time);
```

Comment:
I think this is confusing, from what i see you are sleeping here to simulate the serve time, but we actually put other sellers to sleep when they wait on a condition variable. i think that processing the sale should not actually sleep the thread but simply increment the simulation's clock by one.